resources/qml/InviteDialog.qml

```diff
@@ -0,0 +1,224 @@
+// SPDX-FileCopyrightText: 2021 Nheko Contributors
+//
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+import QtQuick 2.12
+import QtQuick.Controls 2.12
+import QtQuick.Layouts 1.12
+import im.nheko 1.0
+
+ApplicationWindow {
+    id: inviteDialogRoot
+
+    property string roomId
+    property string roomName
+    property InviteesModel invitees
+
+    function addInvite() {
+        if (inviteeEntry.text.match("@.+?:.{3,}")) {
+            invitees.addUser(inviteeEntry.text);
+            inviteeEntry.clear();
+        } else {
+            warningLabel.show();
+        }
+    }
+
+    function cleanUpAndClose() {
+        if (inviteeEntry.text !== "")
+            addInvite();
+        invitees.accept();
+        close();
+    }
+
+    title: qsTr("Invite users to ") + roomName
+    x: MainWindow.x + (MainWindow.width / 2) - (width / 2)
+    y: MainWindow.y + (MainWindow.height / 2) - (height / 2)
+    height: 380
+    width: 340
+
+    Shortcut {
+        sequence: "Ctrl+Enter"
+        onActivated: cleanUpAndClose()
+    }
+
+    Shortcut {
+        sequence: StandardKey.Cancel
+        onActivated: inviteDialogRoot.close()
+    }
+
+    ColumnLayout {
+        anchors.fill: parent
+        anchors.margins: 10
+        spacing: 10
+
+        Label {
+            text: qsTr("User ID to invite")
+            Layout.fillWidth: true
+        }
+
+        RowLayout {
+            spacing: 10
+
+            MatrixTextField {
+                id: inviteeEntry
+
+                backgroundColor: Nheko.colors.window
+                placeholderText: qsTr("@joe:matrix.org", "Example user id. The name 'joe' can be localized however you want.")
+                Layout.fillWidth: true
+                onAccepted: {
+                    if (text !== "") {
+                        addInvite();
+                    }
+                }
+                Component.onCompleted: forceActiveFocus()
+
+                Keys.onShortcutOverride: event.accepted = ((event.key === Qt.Key_Return || event.key === Qt.Key_Enter) && (event.modifiers & Qt.ControlModifier))
+                Keys.onPressed: if ((event.key === Qt.Key_Return || event.key === Qt.Key_Enter) && (event.modifiers === Qt.ControlModifier)) cleanUpAndClose()
+
+            }
+
+            Button {
+                text: qsTr("Add")
+                onClicked: {
+                    if (inviteeEntry.text !== "") {
```

Comment:
Why not set this button to enabled/disabled, if the inviteEntry is empty/does not match the regex?